### PR TITLE
magento/magento2#22010: Updates AbstractExtensibleObject and AbstractExtensibleModel annotations

### PR DESCRIPTION
**Pull request by atwixfirster**
**Issue:** https://github.com/magento/magento2/issues/22010

Thank you!

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=22011) <br/>All committers have signed the CLA.

**Comment by magento-engcom-team** [comment]
Hi @atwixfirster. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by orlangur** [comment]
Not sure it can be delivered this way in a patch release, in the meantime please sign CLA.

**Comment by atwixfirster** [comment]
> in the meantime please sign CLA.

**done**

Thank you, @orlangur

**Comment by atwixfirster** [comment]
@orlangur, any updates here?

Thank you!

**Comment by orlangur** [comment]
@magento run all tests

**Comment by atwixfirster** [comment]
@orlangur , could you please clarify what is a further fate of this PR? 

It has been opened 10 months ago and still have HOLD status. What are we waiting for? :))

Thank you!

**Review by lenaorobei — DISMISSED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lenaorobei, thank you for the review.
[ENGCOM-7071](https://jira.corp.magento.com/browse/ENGCOM-7071) has been created to process this Pull Request
:eight_spoked_asterisk: @lenaorobei, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Comment by engcom-Charlie** [comment]
It should be approved, `Static Tests` failed with error:
> Abstract classes MUST NOT be marked as public @api.

**Comment by lenaorobei** [comment]
Please add ignore annotations to this classes, since it is a recommended way to use `AbstractExtensibleModel` abstract class.

**Comment by atwixfirster** [comment]
> Please add ignore annotations to this classes, since it is a recommended way to use `AbstractExtensibleModel` abstract class.

**done**

File: lib/internal/Magento/Framework/Api/AbstractExtensibleObject.php [hunk 1]
```diff
@@ -12,6 +12,7 @@
  *
  * @SuppressWarnings(PHPMD.NumberOfChildren)
  * @api
+ * @codeCoverageIgnore
```
Comment (lenaorobei) [inline lib/internal/Magento/Framework/Api/AbstractExtensibleObject.php]:
```suggestion
 * phpcs:disable Magento2.Classes.AbstractApi
```

Comment (atwixfirster) [reply lib/internal/Magento/Framework/Api/AbstractExtensibleObject.php]: **done**

@lenaorobei , could you please review? Thank you!

**Review by lenaorobei — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lenaorobei, thank you for the review.
[ENGCOM-7071](https://jira.corp.magento.com/browse/ENGCOM-7071) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Pull request merged**